packages/paste-core/primitives/modal-dialog/stories/index.stories.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from 'react';
+import {storiesOf} from '@storybook/react';
+import {withKnobs} from '@storybook/addon-knobs';
+import styled from '@emotion/styled';
+import {Text} from '@twilio-paste/text';
+import {Button} from '@twilio-paste/button';
+import {ModalDialogOverlay, ModalDialogContent} from '../src';
+
+const StyledModalDialogOverlay = styled(ModalDialogOverlay)({
+  position: 'fixed',
+  top: 0,
+  right: 0,
+  bottom: 0,
+  left: 0,
+  display: 'flex',
+  justifyContent: 'center',
+  alignItems: 'center',
+  background: 'rgba(0, 0, 0, 0.7)',
+});
+
+const StyledModalDialogContent = styled(ModalDialogContent)({
+  width: '100%',
+  maxWidth: '560px',
+  maxHeight: 'calc(100% - 60px)',
+  background: '#f4f5f6',
+  borderRadius: '5px',
+});
+
+interface BasicModalDialogProps {
+  isOpen: boolean;
+  handleClose: () => void;
+}
+
+const BasicModalDialog: React.FC<BasicModalDialogProps> = ({isOpen, handleClose}) => (
+  <StyledModalDialogOverlay isOpen={isOpen} onDismiss={handleClose}>
+    <StyledModalDialogContent>
+      <Text as="p" color="colorText">
+        Roll your own dialog!
+      </Text>
+    </StyledModalDialogContent>
+  </StyledModalDialogOverlay>
+);
+
+const ModalActivator: React.FC = () => {
+  const [isOpen, setIsOpen] = React.useState(false);
+  const handleOpen = (): void => setIsOpen(true);
+  const handleClose = (): void => setIsOpen(false);
+
+  return (
+    <div>
+      <Button variant="primary" onClick={handleOpen}>
+        Open Modal
+      </Button>
+      <BasicModalDialog isOpen={isOpen} handleClose={handleClose} />
+    </div>
+  );
+};
+
+storiesOf('Primitives|ModalDialog', module)
+  .addDecorator(withKnobs)
+  .add('Custom overlay and content', () => {
+    return <ModalActivator />;
```

Comment:
Here's a question: Should we create stories for the main feature of the reach dialog just as a means of knowing what features we need to continue to support if we ever swap out the 3rd party, or if they do a major update and the documentation for the (now, hopefully) widely used features in console etc suddenly disappear.

How much do we care about backwards compat going forward? how much "cover your ass" do we need to make sure consumers of paste and their features still work?

Reply:
Good questions.  

> Should we create stories for the main feature of the reach dialog just as a means of knowing what features we need to continue to support if we ever swap out the 3rd party

I pushed up a slight change so that the story covers every prop on the components we export (Overlay/Content), excluding the normal spread.  I'm not sure which other stories we could add since this primitive is un-styled and intended to be used in only one way.  Let me know what you think may be missing.

>  or if they do a major update and the documentation for the (now, hopefully) widely used features in console etc suddenly disappear.

I've [forked the reach-ui repo](https://github.com/TheSisb/reach-ui) just in case, but we could also leverage git history to recover docs.  This is a fair point and a good reason why we should probably duplicate docs on our doc site rather than pointing to theirs.  I'll do that work in my docsite PR, it's out of scope for this one.

> How much do we care about backwards compat going forward? how much "cover your ass" do we need to make sure consumers of paste and their features still work?

So long as the documentation is viewable, we're fully covered in the fact that we're pinned to a specific version of `@reach/dialog`.  No one will get breaking changes unless we merge it to master.

Reply:
👍